src/bootstrap/src/core/build_steps/run.rs

```diff
@@ -124,19 +124,19 @@ impl Step for Miri {
             std::process::exit(1);
         }
 
-        // This compiler runs on the host, we'll just use it for the target.
-        let target_compiler = builder.compiler(stage, host);
-        let host_compiler = tool::get_tool_rustc_compiler(builder, target_compiler);
+        let compiler = builder.compiler(stage, target);
+        let miri_build = builder.ensure(tool::Miri { compiler, target });
 
         // Get a target sysroot for Miri.
-        let miri_sysroot = test::Miri::build_miri_sysroot(builder, target_compiler, target);
+        let miri_sysroot =
+            test::Miri::build_miri_sysroot(builder, miri_build.target_compiler, target);
```

Comment:
Question: wait hold on, previously, this was the *host* compiler (`builder.compiler(stage, host)`). ~~But now, we're building the miri sysroot with the *target* compiler.~~

EDIT: wait no, I think the `ToolBuildResult::target_compiler` is what would get passed to `Step`s... Maybe we should rename that field as `step_compiler` or sth, since it's not about the compiler that would be used for the target platform...? Or am I misreading it?

Reply:
I am not sure how `step_compiler` helps to clarify the meaning. :thinking: Having two fields named `build_compiler` and `target_compiler` seems straightforward to me. Also, I think renaming that type shouldn't be part of this PR.